### PR TITLE
feat: allow optional function to be used to modify and filter out unneeded log files

### PR DESCRIPTION
**Pull request by blacha**

### CODE REVIEW

**Review by daveball-linz — APPROVED** [review]

File: src/shipper/index.ts [hunk 1]
```diff
@@ -2,3 +2,15 @@ export { Log, version, hash } from '../logger';
...
+// Inject additional log functions if there some provided
+import { LogShipper } from './shipper.config';
+import { LogFunctionName } from './type';
+try {
+  const AdditionalLogFuncs = require(`./${LogFunctionName}`) ?? [];
+  console.log('FoundAdditionalFunctions', AdditionalLogFuncs);
+  for (const fn of AdditionalLogFuncs) LogShipper.DefaultLogProcessFunctions.push(fn);
+} catch (e) {
+  //ignore
+}
```
Comment (jacott) [inline src/shipper/index.ts:16]: Could add this
```typescript
if (e.code !== 'MODULE_NOT_FOUND') {
  throw e;
}
```

File: src/shipper/log.funcs/extract.json.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export function onLogExtractJson(lo: LogObject): boolean | void {
+  if (lo.message == null) return;
+
+  const extracted = extractJson(lo.message);
+  if (extracted == null) return;
+
+  for (const [key, value] of Object.entries(extracted)) {
+    lo[key] = value;
+  }
```
Comment (jacott) [inline src/shipper/log.funcs/extract.json.ts:31]: Why not `Object.assign(lo, extracted)`?

File: src/shipper/log.funcs/extract.json.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  delete lo.message; // JSON logs don't need the raw JSON in the output
+  return;
```
Comment (jacott) [inline src/shipper/log.funcs/extract.json.ts:34]: more efficient to write `lo.message = undefined`

File: src/shipper/index.ts (hunk 1 above)
Comment (blacha) [reply src/shipper/index.ts:16]: Good idea

Comment (blacha) [reply src/shipper/index.ts:16]: Adding this, throws a `ENOENT` and just catching that throws a different error, might be easier to just throw

**Review by jacott — APPROVED** [review]

**Pull request merged**